clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/CloudFormationController.java

```diff
@@ -14,42 +14,51 @@
  * limitations under the License.
  */
 
-package com.netflix.spinnaker.clouddriver.controllers
+package com.netflix.spinnaker.clouddriver.controllers;
 
-import com.netflix.spinnaker.clouddriver.model.CloudFormation
-import com.netflix.spinnaker.clouddriver.model.CloudFormationProvider
-import groovy.util.logging.Slf4j
-import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.data.rest.webmvc.ResourceNotFoundException
-import org.springframework.web.bind.annotation.*
+import com.netflix.spinnaker.clouddriver.model.CloudFormation;
+import com.netflix.spinnaker.clouddriver.model.CloudFormationProvider;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Slf4j
 @RequestMapping("/cloudFormation")
 @RestController
 class CloudFormationController {
 
   @Autowired
-  List<CloudFormationProvider> cloudFormationProviders
+  private List<CloudFormationProvider> cloudFormationProviders;
 
   @RequestMapping(method = RequestMethod.GET, value = "/list/{accountId}")
   List<CloudFormation> list(@PathVariable String accountId,
-                            @RequestParam(required = false, defaultValue = '*') String region) {
-    log.debug("Cloud formation list stacks for account $accountId")
-    cloudFormationProviders.collect {
-      it.list(accountId, region)
-    }.flatten()
+                            @RequestParam(required = false, defaultValue = "*") String region) {
+    log.debug("Cloud formation list stacks for account $accountId");
+    return cloudFormationProviders
+      .stream()
+      .map(p -> p.list(accountId, region))
+      .flatMap(Collection<CloudFormation>::stream)
+      .collect(Collectors.toList());
   }
 
   @RequestMapping(method = RequestMethod.GET, value = "/get")
   CloudFormation get(@RequestParam String stackId) {
-    log.debug("Cloud formation get stack with id $stackId")
-    try {
-      cloudFormationProviders.collect {
-        it.get(stackId)
-      }.first()
-    } catch (NoSuchElementException e) {
-      throw new ResourceNotFoundException("Cloud Formation stackId $stackId not found.")
-    }
+    log.debug("Cloud formation get stack with id $stackId");
+    return cloudFormationProviders
+      .stream()
+      .map(p -> p.get(stackId))
+      .filter(Optional::isPresent)
+      .map(Optional<CloudFormation>::get)
+      .findFirst()
+      .<ResourceNotFoundException>orElseThrow(() -> {
```

Comment:
I don't think you need the generic stuff here, just do `.orElseThrow(() -> {`

Reply:
I think you can just do this? `orElseThrow(() -> new ResourceNotFoundException("..."));`

Reply:
👍 Done in https://github.com/spinnaker/clouddriver/pull/3279/commits/1e91224e4c0d384adb0ff8ddeb192a120b7411c0